examples/call-runtime/lib.rs

```diff
@@ -0,0 +1,222 @@
+        use ink_e2e::build_message;
+
+        type E2EResult<T> = Result<T, Box<dyn std::error::Error>>;
+
+        /// The contract will be given 1000 tokens during instantiation.
+        const CONTRACT_BALANCE: Balance = 1_000_000_000_000_000;
+        /// The receiver will get enough funds to have the required existential deposit.
+        ///
+        /// If your chain has this threshold higher, increase the transfer value.
+        const TRANSFER_VALUE: Balance = 1_000_000_000;
+
+        #[ink_e2e::test]
+        #[ignore = "Requires that the pallet contract is configured with:\
+            - `CallFilter` allowing for a transfer, e.g. `frame_support::traits::Everything`,\
+            - `UnsafeUnstableInterface = ConstBool<true>`"]
+        async fn transfer_with_call_runtime_works(
+            mut client: Client<C, E>,
+        ) -> E2EResult<()> {
+            // given
+            let constructor = RuntimeCallerRef::new();
+            let contract_acc_id = client
+                .instantiate(
+                    "call-runtime",
+                    &ink_e2e::alice(),
+                    constructor,
+                    CONTRACT_BALANCE,
+                    None,
+                )
+                .await
+                .expect("instantiate failed")
+                .account_id;
+
+            let receiver: AccountId = default_accounts::<DefaultEnvironment>().bob;
+
+            let contract_balance_before = client
+                .balance(contract_acc_id)
+                .await
+                .expect("Failed to get account balance");
+            let receiver_balance_before = client
+                .balance(receiver)
+                .await
+                .expect("Failed to get account balance");
+
+            // when
+            let transfer_message = build_message::<RuntimeCallerRef>(contract_acc_id)
+                .call(|caller| caller.transfer_through_runtime(receiver, TRANSFER_VALUE));
+
+            let call_res = client
+                .call(&ink_e2e::alice(), transfer_message, 0, None)
+                .await
+                .expect("call failed");
+
+            assert!(call_res.dry_run.exec_result.result.is_ok());
+
+            // then
+            let contract_balance_after = client
+                .balance(contract_acc_id)
+                .await
+                .expect("Failed to get account balance");
+            let receiver_balance_after = client
+                .balance(receiver)
+                .await
+                .expect("Failed to get account balance");
+
+            assert_eq!(
+                contract_balance_before,
+                contract_balance_after + TRANSFER_VALUE
+            );
+            assert_eq!(
+                receiver_balance_before,
+                receiver_balance_after - TRANSFER_VALUE
+            );
+
+            Ok(())
+        }
+
+        /// In the standard configuration, the node doesn't allow for `call-runtime` usage.
+        #[ink_e2e::test]
```

Comment:
And then here I'd have the feature flag toggle this as well

Reply:
`not(feature = "permissive-node")`